docs/user-guide/reduction-workflow-guidelines.md

```diff
@@ -0,0 +1,139 @@
+# Reduction Workflow Guidelines
+
+## Introduction
+
+TODO
+
+- Package and module structure (where to place types? what goes where?)
+  - Requires minor discussion
+- Loading from SciCat vs. local files (e.g., define run ID, choose provider that either converts to local path, or uses service to get file and return path)
+- Should we have default params set in workflows?
+  - Avoid unless good reason.
+  - Can have widgets that generate dict of params and values, widgets can have defaults
+
+## Nomenclature
+
+- *Provider*: A callable step in a workflow writing with Sciline.
+
+## D: Documentation
+
+### D.1: Document math and references in docstrings
+
+**Reason**
+Documentation should be as close to the code as possible, to decrease the chance that it runs out of sync.
+This includes mathematical formulas and references to literature.
+
+**Note**
+We have previously documented math and references in Jupyter notebooks.
+This is not sufficient, as the documentation is not close to the code.
+
+## N: Naming
+
+TODO
+- Naming conventions (and type conventions (example: filenames)?)
+
+## P: Performance
+
+### P.1: Runtime and memory use of workflows shall be tested with large data
+
+**Reason**
+We want to ensure that the workflows are efficient and do not consume excessive memory.
+
+**Note**
+This is often not apparent from small test data, as the location of performance bottlenecks may depend on the size of the data.
+
+## S: Structure
+
+### S.1: Workflows shall be able to return final results as event data
+
+**Reason**
+- Required for polarization analysis, which wraps a base workflow.
+- Required for subsequent filtering, unless part of the workflow.
+
+**Note**
+There should be a workflow parameters (flag) to select whether to return event data or not.
+
+### S.2: Split loading and handling of monitors and detectors from loading of auxiliary data and metadata
+
+**Reason**
+- Allows for more efficient parallelism and reduction in memory use.
+- Avoids loading large data into memory that is not needed for the reduction.
+- Avoids keeping large data alive in memory if output metadata extraction depends on auxiliary input data or input metadata.
+
+### S.3: Avoid dependencies of output metadata on large data
+
+**Reason**
+Adding dependencies on large data to the output metadata extraction may lead to large data being kept alive in memory.
+
+**Note**
+Most of this is avoid by following S.2.
+A bad exampleAvoid would be to write to total raw counts to the output metadata, as this would require keeping the large data alive in memory, unless it is ensured that the task runs early.
```

Comment:
```suggestion
A bad example would be writing the total raw counts to the output metadata, as this would require keeping the large data alive in memory, unless it is ensured that the task runs early.
```